pkg/cmd/container/exec.go

```
@@ -133,13 +133,10 @@ func execActionWithContainer(ctx context.Context, client *containerd.Client, con
 		return nil
 	}
 	status := <-statusC
-	code, _, err := status.Result()
+	_, _, err = status.Result()
 	if err != nil {
 		return err
 	}
-	if code != 0 {
```

@zouyee not displaying the message is one thing, but clearly we still need to error out.

---

The executable we ran inside the container that errored, we need to print error ?

---

We need to return an error from this method, otherwise nerdctl will exit 0.

---

Why does exec function care about the results of execution inside the container?

---

It should be just same as Docker

---

nerdctl exec
```
nerdctl -n k8s.io exec  xxx bash -c  'bash ./faketime.sh xxx';echo $?
tomcat: stopped
tomcat: started
FATA[0005] exec failed with exit code 1
1
```

docker exec
```
docker exec xxx  bash -c  'bash ./faketime.sh xxx';echo $?
tomcat: stopped
tomcat: started
1
```

---

Yes, we should do the same thing as docker.
With your patch right now, we no longer return exit 1 because you do not return a go error.

---

Agree with you @zouyee  you may write stderr to `/dev/null` as workaround